ambry-store/src/main/java/com.github.ambry.store/StorageManager.java

```diff
@@ -157,16 +159,30 @@ public void start() throws InterruptedException {
     }
   }
 
-  /**
-   * @param id the {@link PartitionId} to find the store for.
-   * @return the {@link Store} corresponding to the given {@link PartitionId}, or {@code null} if no store was found for
-   *         that partition, or that store was not started.
-   */
+  @Override
   public Store getStore(PartitionId id) {
     DiskManager diskManager = partitionToDiskManager.get(id);
     return diskManager != null ? diskManager.getStore(id) : null;
   }
 
+  @Override
+  public ServerErrorCode isPartitionAvailable(PartitionId partition, ReplicaId localReplica) {
+    if (getStore(partition) == null) {
+      if (localReplica != null) {
+        // check stores on the disk
+        if (!isDiskAvailable(localReplica.getDiskId())) {
+          localReplica.markDiskDown();
```

Comment:
Looks like you're doing this twice.  Better to let caller do it since side effects are bad.

Reply:
fixed.